### PR TITLE
Aud claim addition in backend JWTs

### PR DESCRIPTION
**Pull request by slahirucd7**
### Purpose
This PR supports to add `aud` claim to the backend JWTs.
In the `api.yaml` file users can define the relevant backJWT audience values and those values.
During the API invocation time those user defined audience claims will be added to the backend JWT.

### Issues

Fixes #

### Automation tests
 - Unit tests added: No
 - Integration tests added: Yes

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

File: api/proto/wso2/discovery/api/backend_jwt_configuration.proto [hunk 1]
```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2023, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (renuka-fernando) [inline api/proto/wso2/discovery/api/backend_jwt_configuration.proto:2]:
```suggestion
 * Copyright (c) 2023, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
```

File: integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/testcases/standalone/jwtGenerator/BackendJWTWithAudClaimTestcase.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2023, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
```
Comment (renuka-fernando) [inline integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/testcases/standalone/jwtGenerator/BackendJWTWithAudClaimTestcase.java:4]:
```suggestion
 * Copyright (c) 2023, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
 *
 * WSO2 LLC. licenses this file to you under the Apache License,
```

File: integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/testcases/standalone/jwtGenerator/BackendJWTWithAudClaimTestcase.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        Assert.assertTrue(respHeaders.containsKey(ResponseConstants.BACKEND_JWT_DEFAULT_HEADER_NAME),
+                "Backend JWT relevant header not found in the response");
+        String backendJWT = respHeaders.get(ResponseConstants.BACKEND_JWT_DEFAULT_HEADER_NAME);
+        SignedJWTInfo signedJWTInfo = JWTUtils.getSignedJwt(backendJWT);
+        JWTClaimsSet jwtClaimsSet = signedJWTInfo.getJwtClaimsSet();
+        Map<String, Object> jwtClaimSet = jwtClaimsSet.getClaims();
+        Assert.assertNotNull(jwtClaimSet, "Cannot find JWT claim set in the backend JWT");
+        Assert.assertTrue(jwtClaimSet.containsKey("aud"), "Cannot find aud claim in the backend JWT");
+        List<String> audList = (List<String>) jwtClaimSet.get("aud");
+        Assert.assertTrue(audList.get(0).equalsIgnoreCase("https://petstore.swagger.io")
```
Comment (renuka-fernando) [inline integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/testcases/standalone/jwtGenerator/BackendJWTWithAudClaimTestcase.java:65]: Do we want to ignore case here?

**Review by renuka-fernando — APPROVED** [review]

**Review by renuka-fernando — DISMISSED** [review]

**Review by renuka-fernando — APPROVED** [review]

**Review by renuka-fernando — APPROVED** [review]

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(EastUS) cluster : dev-deployment-v2 : 20230905.39

**Comment by choreo-cicd** [comment]
[succeeded]  : dev-deployment-v2 : 20230905.39

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(NorthEU) cluster : dev-deployment-v2 : 20230905.39

**Comment by choreo-cicd** [comment]
[succeeded] Controlplane cluster : dev-deployment-v2 : 20230905.39

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(NorthEU) cluster : prod-deployment-v2 : 20230907.5

**Comment by choreo-cicd** [comment]
[succeeded] Controlplane cluster : prod-deployment-v2 : 20230907.5

**Pull request merged**